src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsNuGetPathContextFactory.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnvDTE;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.ProjectManagement;
+using NuGet.ProjectManagement.Projects;
+using NuGet.ProjectModel;
+
+namespace NuGet.VisualStudio
+{
+    [Export(typeof(IVsNuGetPathContextFactory))]
+    public class VsNuGetPathContextFactory : IVsNuGetPathContextFactory
+    {
+        private readonly ISettings _settings;
+        private readonly IVsSolutionManager _solutionManager;
+
+        [ImportingConstructor]
+        public VsNuGetPathContextFactory(ISettings settings, IVsSolutionManager solutionManager)
+        {
+            _settings = settings;
+            _solutionManager = solutionManager;
+        }
+
+        /// <summary>
+        /// This property is used only for testing.
+        /// </summary>
+        public Func<BuildIntegratedNuGetProject, Task<LockFile>> GetLockFileOrNullAsync { get; set; }
```

Comment:
Ew...
I'd suggest adding a separate constructor with 3rd parameter for testing purposes. Plus in regular constructor you can set its value to `BuildIntegratedProjectUtility.GetLockFileOrNull` right away.